libs/libc/stdlib/lib_cxa_atexit.c

```diff
@@ -0,0 +1,56 @@
+/****************************************************************************
+ * libs/libc/stdlib/lib_cxa_atexit.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <nuttx/atexit.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: __cxa_atexit
+ *
+ * Description:
+ *   __cxa_atexit() registers a destructor function to be called by exit().
+ *   On a call to exit(), the registered functions should be called with
+ *   the single argument 'arg'. Destructor functions shall always be
+ *   called in the reverse order to their registration (i.e. the most
+ *   recently registered function shall be called first),
+ *
+ *   If shared libraries were supported, the callbacks should be invoked
+ *   when the shared library is unloaded as well.
+ *
+ * Reference:
+ *   Linux base
+ *
+ ****************************************************************************/
+
+int __cxa_atexit(CODE void (*func)(FAR void *), FAR void *arg,
```

Comment:
do we need to re-evaluate `libs/libxx/libxx_eabi_atexit.cxx`?

Reply:
Yes, @pkarashchenko it's here:
https://github.com/apache/incubator-nuttx/pull/6197/commits/d9c38f6d0a583d1202ad2bbf187cbf70b54157c4#diff-1dd7ea5d4a2882c6bc7400e18c5ae43d9c27bc69067ca6d86a7344ad6ad274aaR45

Reply:
I still see `CXXSRCS += libxx_cxa_atexit.cxx` in `Makefile` and `libs/libxx/libxx_eabi_atexit.cxx` seems still be present in the source tree... I was expecting it can be removed? Am I missing something?

Reply:
My bad. Already found it

Reply:
Ok. Let's remove `libs/libxx/libxx.hxx` as well. Seems to be not needed anymore

Reply:
Not quite true, that is used to declare __cxa_atexit()

Reply:
_cxa_atexit is called by compiler internally not by application directly, so it could be removed safely.